datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/DatalakeHorizontalScaleActions.java

```diff
@@ -0,0 +1,213 @@
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleCmRollingRestartEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleCmRollingRestartRequest;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleFailedEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleFinishedEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleStartEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleStartRequest;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleValidationFailedEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.event.DatalakeHorizontalScaleValidationStartEvent;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.AbstractAction;
+import com.sequenceiq.flow.core.CommonContext;
+import com.sequenceiq.flow.core.FlowParameters;
+
+@Configuration
+public class DatalakeHorizontalScaleActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeHorizontalScaleActions.class);
+
+    private final SdxService sdxService;
+
+    private final SdxStatusService sdxStatusService;
+
+    public DatalakeHorizontalScaleActions(SdxService sdxService, SdxStatusService sdxStatusService) {
+        this.sdxService = sdxService;
+        this.sdxStatusService = sdxStatusService;
+    }
+
+    @Bean(name = "DATALAKE_HORIZONTAL_SCALE_VALIDATION_STATE")
+    public Action<?, ?> datalakeHorizontalScaleValidationStart() {
+        return new AbstractSdxAction<>(DatalakeHorizontalScaleValidationStartEvent.class) {
+            @Override
+            protected void doExecute(SdxContext context, DatalakeHorizontalScaleValidationStartEvent payload, Map<Object, Object> variables) {
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.DATALAKE_HORIZONTAL_SCALE_ON_DATALKE_VALIDATION_IN_PROGRESS,
+                        String.format("Validation of horzontal scale is in progress for group of %s with instance count %s on the Data Lake",
+                                payload.getStackScaleRequest().getGroup(),
+                                payload.getStackScaleRequest().getDesiredCount()),
```

Comment:
Is required validation in the number of instances? what happened if the input for Core HG is less them 3 instances?